airbyte-ci/connectors/connector_ops/connector_ops/utils.py

```diff
@@ -260,35 +262,53 @@ def _get_type_and_name_from_technical_name(self) -> Tuple[str, str]:
         name = self.technical_name[len(_type) + 1 :]
         return _type, name
 
+    @property
+    def technical_name(self) -> str:
+        """
+        Return the technical name of the connector from the given _relative_connector_path
+        e.g. source-google-sheets -> google-sheets or third-party/farosai/airbyte-pagerduty-source -> airbyte-pagerduty-source
+        """
+        return self._relative_connector_path.split("/")[-1]
+
     @property
     def name(self):
         return self._get_type_and_name_from_technical_name()[1]
 
     @property
     def connector_type(self) -> str:
-        return self._get_type_and_name_from_technical_name()[0]
+        return self.metadata["connectorType"]
+
+    @property
+    def is_third_party(self) -> bool:
+        return THIRD_PARTY_GLOB in self._relative_connector_path
 
     @property
     def documentation_directory(self) -> Path:
+        if self.has_external_documentation:
+            return None
         return Path(f"./docs/integrations/{self.connector_type}s")
```

Comment:
This `.docs` is one of the reason why airbyte-ci is required to run from the repo root. For more flexibility we shall add a constant like `GIT_REPO_ROOT = git.Repo(search_parent_directories=True)` and build path from this root.

Reply:
So I would have love to do this inside this PR but it got a bit big so I broke it out into this branch
https://github.com/airbytehq/airbyte/pull/31409

Though it definately raises the question of if we would want to do this way....

The alternative is just having `airbyte-ci` set the working directory to the repo root 🤔